chain-signatures/node/src/protocol/presignature.rs

```diff
@@ -157,8 +157,13 @@ impl PresignatureManager {
     }
 
     pub fn garbage_collect(&mut self, cfg: &ProtocolConfig) {
+        let before = self.gc.len();
         self.gc
             .retain(|_, instant| instant.elapsed() < Duration::from_millis(cfg.garbage_timeout));
+        let removed = before.saturating_sub(self.gc.len());
+        if removed > 0 {
+            tracing::warn!("garbage collected {} presignatures", removed);
```

Comment:
I wouldn't say this is a warning though, since this is purely informational where warning should be reserved for logs that should close to never happen. So it should be `info` instead.

Reply:
I think we should minimize the number of garbage collections since it is a waste and an indicator that our system is not efficient.

Reply:
I think you're misunderstanding what garbage collection is. Garbage collections happens no matter what, as in every single triple, presignature and signature will eventually be garbage collected and this will always be called. It is our way to keep tally of what IDs to keep around until messages stop from other protocol that linger behind or are delayed due to restarts or other misc behaviors

Reply:
kk, I got it
what I meant is that we should keep track of gcd triples, pre-signatures that were not used for any reason